airflow/providers/databricks/hooks/databricks.py

```diff
@@ -490,6 +567,41 @@ def start_cluster(self, json: dict) -> None:
         """
         self._do_api_call(START_CLUSTER_ENDPOINT, json)
 
+    def activate_cluster(self, json: dict, polling: int, timeout: int | None = None) -> None:
+        """
+        Start the cluster, and wait for it to be ready.
+
+        :param json: json dictionary containing cluster specification.
+        :param polling: polling interval in seconds.
+        :param timeout: timeout in seconds.
+        """
+        cluster_id = json["cluster_id"]
+
+        api_called = False
+        time_start = time.time()
+
+        while True:
```

Comment:
I think this one is already implemented in start_cluster method which is [line 562](https://github.com/apache/airflow/blob/24c003e8435cbfcafbdc714f48e9d03410b8cb11/airflow/providers/databricks/hooks/databricks.py#L562)

Reply:
If that's the case, should we consolidate them into one method? For me, I can not understand the logic difference by reading the names `start_cluster` and `activate_cluster`. But this's not a huge issue though.

Reply:
When start a Databricks Cluster, it tunrs on as `PENDING` state, so we cannot use the cluster until it's on ready state. That's the what `start_cluster()` method does.

Meanwhile, `active_cluster()` method guarantees the cluster is on `Running` State, so after the completed, we can utilize the cluster immediately.

Sometimes users want just starting the cluster, or want making the cluster into the usable status.

Reply:
Yep, I know it's a desired behavior. Just wondering if it would be possible for us to merge the method and toggle the waiting behavior through a parameter

Reply:
Yes, maybe. I now understand your suggestion. I think calling`active_cluster()` method without `polling` and `timeout` is same as calling `start_cluster()`. So, if I change the code it looks like this

```py
def start_cluster(self, json: dict, polling: int | None = None, timeout: int | None = None) -> None:
        cluster_id = json["cluster_id"]

        api_called = False
        time_start = time.time()

        while True:
            # Here is changed
            # run_state = self.get_cluster_state(cluster_id)
            run_state = self._do_api_call(START_CLUSTER_ENDPOINT, json)

            if run_state.is_running:
                return
            elif run_state.is_terminal:
                if api_called:
                    raise AirflowException(
                        f"Cluster {cluster_id} start failed with '{run_state.state}' "
                        f"state: {run_state.state_message}"
                    )

                self.start_cluster(json)
                api_called = True

            # wait for cluster to start
            time.sleep(polling)

            elapsed_time = time.time() - time_start
            if timeout and elapsed_time > timeout:
                raise AirflowException(f"Cluster {cluster_id} start timed out after {timeout} seconds")
```

Please, take a look and get some opinions 🙏 


\+ I think then we can apply this to `restart_cluster()` method too!

Reply:
I would prefer to have one method only, with default to return immediately, but allow to wait until start/rester when specifying options.